src/components/Formatters/useCurrencyFormatter.tsx

```diff
@@ -0,0 +1,18 @@
+export enum CurrencypPosition {
+  BACK = 'de-DE',
+  FRONT = 'ja-JP',
+}
+
+export const useCurrencyFormatter = (
+  options?: Intl.NumberFormatOptions & { currencyFormat?: CurrencypPosition }
+) => {
+  const formatter = new Intl.NumberFormat(
+    options?.currencyFormat ?? CurrencypPosition.FRONT,
+    {
+      style: options?.style ?? 'currency',
+      currency: options?.currency ?? 'USD',
```

Comment:
Spreading `options` below this will override this value if it's given in options so there's no need to try to use it here.
```suggestion
      style: 'currency',
      currency: 'USD',
```